developer-docs-site/docs/move/move-on-aptos/cryptography.md

```diff
@@ -0,0 +1,186 @@
+3. Verifying an Ed25519 signature via (the now obsolete) [`signature::ed25519_verify`](https://github.com/aptos-labs/aptos-core/blob/3b53225b5a1effcce5dee9597a129216510dc424/aptos-move/framework/aptos-framework/sources/Signature.move#L19)
+4. Validating an Ed25519 public key via (the now obsolete) [`signature::ed25519_validate_pubkey`](https://github.com/aptos-labs/aptos-core/blob/3b53225b5a1effcce5dee9597a129216510dc424/aptos-move/framework/aptos-framework/sources/Signature.move#L9)
+
+Since then, we have heavily extended support for cryptography in Move. 
+Behind these new cryptographic features were two **key guiding principles**:
+
+1. **Minimize gas costs** for Move developers by implementing key primitives as [Move native functions](./move/book/functions#native-functions) (e.g., see the [`aptos_std::bls12381`](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-stdlib/sources/cryptography/bls12381.move) module for BLS signatures over BLS12-381 elliptic curves) 
+2. When native functions are not available, **empower developers** to implement their own cryptographic primitives on top of abstract cryptographic building blocks such as _finite fields_ and _Abelian groups_ (e.g., via the [`aptos_std::crypto_algebra`](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-stdlib/sources/cryptography/crypto_algebra.move) module).
+
+An overview of these features and (some of) the applications they enable follows below. 
+For a detailed view, see the [cryptography Move modules code](https://github.com/aptos-labs/aptos-core/tree/main/aptos-move/framework/aptos-stdlib/sources/cryptography).
+
+## Cryptographic hash functions in Move
+
+Developers can now use more cryptographic hash functions in Move via the [`aptos_std::aptos_hash`](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-stdlib/sources/hash.move) module:
+
+ 1. Keccak256
+ 2. SHA2-512
+ 3. SHA3-512
+ 4. RIPEMD160
+ 5. Blake2b-256
+
+Some of these functions can be used for interoperability with other chains (e.g., verifying Ethereum Merkle proofs via [`aptos_std::aptos_hash::keccak256`](https://github.com/aptos-labs/aptos-core/blob/137acee4c6dddb1c86398dce25b041d78a3028d3/aptos-move/framework/aptos-stdlib/sources/hash.move#L35)).
+Others, have lower gas costs, such as [`aptos_std::aptos_hash::blake2b_256`](https://github.com/aptos-labs/aptos-core/blob/137acee4c6dddb1c86398dce25b041d78a3028d3/aptos-move/framework/aptos-stdlib/sources/hash.move#L69).
+In general, a wider variety of hash functions give developers additional freedom in terms of both security and interoperability with other off-chain cryptographic system.
+
+## Digital signature verification in Move
+
+Developers can now use a *type-safe* API for verifying many types of digital signatures in Move:
```

Comment:
typesafety seems to be a core priniciple.

Reply:
Fixed: added it to key guiding principles.